picard/metadata.py

```diff
@@ -300,6 +300,8 @@ def compare_to_track(self, track, weights):
 
         result = SimMatchTrack(similarity=-1, releasegroup=None, release=None, track=None)
         for release in releases:
+            if 'medium-count' not in release and 'media' in release:  # generate this field for MusicBrainz, it is already part of acoustid
+                release['medium-count'] = len(release['media'])
```

Comment:
Do I miss something here? `release['medium-count']` is not used in the code here, why is it set here? The problem I see with this is that it changes the data passed to `compare_to_track`, which is an unwanted side effect. If we want to make sure `release['medium-count']` is set it should be done elsewhere (e.g. after loading the data), but not here.

Reply:
ok to skip these 2 lines, we can re-visit in future if needed.

Reply:
And in thinking about it, yes I agree, the medium-count should not be added as part of compare_to_track().  as thats not its purpose.  If we want to add this later, I'm open to suggestions on where to put it.

Reply:
Ok, I merged it without those lines. I need to look at where we best can fit this if needed.